# $Id$
#
# @TEST-EXEC-FAIL: hilti-build2 %INPUT -o a.out >output 2>&1
# @TEST-EXEC:      btest-diff output

module Main

import Hilti

type myException = exception<string>

void run() {
     local ref<myException> e
     e = new myException
     exception.throw e
}

# @TEST-START-NEXT

module Main

import Hilti

type myException = exception<string>

void run() {
     local ref<myException> e
     e = new myException 42
     exception.throw e
}

# @TEST-START-NEXT

module Main

import Hilti

type myException = exception

void run() {
     local ref<myException> e
     e = new myException "hello"
     exception.throw e
}
